Entities/Natural/Animals/Chicken/IncubateEgg.as

```diff
@@ -0,0 +1,35 @@
+#include "EmotesCommon.as"
+#include "CrouchCommon.as"
+
+
+void onInit(CBlob@ this) {
+	this.getCurrentScript().tickFrequency = 10;
+}
+
+void onTick(CBlob@ this) {
+    if (this.hasTag("is_incubating_egg")) {
+        set_emote(this, 51);  // do chicken emote
+
+        if (!isIncubatingEgg(this)) {
+            this.Untag("is_incubating_egg");
+            set_emote(this, Emotes::off);
+        }
+    }
+}
+
+/* Check if we're incubating an egg. */
+bool isIncubatingEgg(CBlob@ this) {
+    if (!isCrouching(this))
+        return false;
+
+	CBlob@[] overlapping;
+	this.getOverlapping(@overlapping);
+
+	for (int i=0; i < overlapping.length; ++i) {
+		CBlob@ blob = overlapping[i];
+		if (blob.getName() == "egg") {
+			return true;
+		}
+	}
+	return false;
+}
```

Comment:
Overall seems good, but please don't forget newline at end of files. Can be merged and done manually though.